synapse/handlers/receipts.py

```diff
@@ -91,13 +91,23 @@ async def _received_remote_receipt(self, origin: str, content: JsonDict) -> None
                         )
                         continue
 
+                    # Check if these receipts apply to a thread.
+                    thread_id = None
+                    data = user_values.get("data", {})
+                    if isinstance(data, dict):
+                        thread_id = data.get("thread_id")
+                        # If the thread ID is invalid, consider it missing.
+                        if not isinstance(thread_id, str):
+                            thread_id = None
```

Comment:
Do we not want to gate this behind `msc3771_enabled` too?

Reply:
Um, yeah I guess that makes sense?